src/Apollo/NcrReactionsProjector.php

```diff
@@ -0,0 +1,216 @@
+        $tableName = $this->tableManager->getNodeTableName($reactionsRef->getQName(), $context);
+
+        foreach ($event->get('reactions') as $reaction) {
+            if ($reactions->isInMap('reactions', $reaction)) {
+                $updateExpression .= " reactions.#{$reaction} = reactions.#{$reaction} + :v_incr,";
+                $expressionAttributeNames["#{$reaction}"] = $reaction;
+            }
+        }
+
+        if (empty($expressionAttributeNames)) {
+            return;
+        }
+
+        $params = [
+            'TableName'                 => $tableName,
+            'Key'                       => [
+                NodeTable::HASH_KEY_NAME => ['S' => $reactionsRef->toString()],
+            ],
+            'UpdateExpression'          =>  rtrim('set'. $updateExpression, ', '),
+            'ExpressionAttributeNames'  => $expressionAttributeNames,
+            'ExpressionAttributeValues' => [
+                ':v_incr' => ['N' => '1'],
+            ],
+            'ReturnValues'              => 'NONE',
+        ];
+
+        $this->client->updateItem($params);
+        $this->ncrSearch->indexNodes([$this->ncr->getNode($reactionsRef, true, $context)], $context);
+    }
+
+    protected function projectNode(Message $reactions, Message $event, Pbjx $pbjx): void
+    {
+        $context = ['causator' => $event];
+        $reactions
+            ->set('updated_at', $event->get('occurred_at'))
+            ->set('updater_ref', $event->get('ctx_user_ref'))
+            ->set('last_event_ref', $event->generateMessageRef())
+            ->set('etag', Aggregate::generateEtag($reactions));
+
+        $this->ncr->putNode($reactions, null, $context);
+        $this->ncrSearch->indexNodes([$reactions], $context);
+        $pbjx->trigger($reactions, 'projected', new NodeProjectedEvent($reactions, $event), false, false);
+    }
+
+    protected function createReactions(NodeRef $nodeRef): Message
+    {
+        static $class = null;
+        if (null === $class) {
+            $class = MessageResolver::resolveCurie('*:apollo:node:reactions:v1');
+        }
+
+        $reactions = $class::fromArray(['_id' => $nodeRef->getId()]);
+        $this->addReactionTypes($reactions);
+
+        return $reactions;
+    }
+
+    protected function getReactions(NodeRef $nodeRef, Message $event): ?Message
+    {
+        try {
+            $reactionsRef = $this->createReactionsRef($nodeRef);
+            return $this->ncr->getNode($reactionsRef, true, ['causator' => $event]);
+        } catch (NodeNotFound $nf) {
+            // Node not found do nothing here
+        } catch (\Throwable $e) {
+            throw $e;
+        }
+
+        return null;
+    }
+
+    protected function addReactionTypes(Message $reactions): void
+    {
+        foreach ([
+                     'love',
+                     'haha',
+                     'wow',
+                     'wtf',
+                     'trash',
+                     'sad',
```

Comment:
@gdbrown @bhsiao55 if the assumption is that each site will have its own set of reactions, i'd probably have an empty array at triniti level or a separate method that returns an empty array

Reply:
@bhsiao55 @efjacobson i'd create a validator to ensure we reject any commands with invalid reactions.  this class doesn't need to do any validation or filtering of reaction types.

Reply:
@efjacobson I thought we were not going to allow customization at the site level but If you want to go that route I can change it.  What do you think @gdbrown?

Reply:
@bhsiao55 @efjacobson could have a common set in a validator but allow override.  could even simply filter them, i.e. remove ones not allowed rather than throw an exception since someone doing that probably isn't using the interface.  ReactionsValidator::validateAddReactions or something.  but mainly the projector isn't concerned with what the reactions are, it just updates the counts.

Reply:
@gdbrown @efjacobson added ReactionsValidator